crates/bevy_ecs/src/system/system_registry.rs

```diff
@@ -43,6 +43,29 @@ pub struct SystemId<I = (), O = ()> {
     pub(crate) marker: std::marker::PhantomData<fn(I) -> O>,
 }
 
+impl<I, O> SystemId<I, O> {
+    /// Transforms a [`SystemId`] into the [`Entity`] that holds the one-shot system's state.
+    ///
+    /// It's trivial to convert [`SystemId`] into an [`Entity`] since a system
+    /// is really an entity with associated handler function.
+    ///
+    /// For example, this is useful if you want to assign a name label to a system.
+    pub fn entity(self) -> Entity {
+        self.entity
+    }
+
+    /// Create [`SystemId`] from an [`Entity`]. Useful when you only have entity handles to avoid
+    /// adding extra components that have a [`SystemId`] everywhere. To run a system with this ID
+    ///  - The entity must be a system
+    ///  - The `I` + `O` types must be correct
+    pub fn from_entity(entity: Entity) -> Self {
```

Comment:
Can you use this method here [in this impl](https://github.com/bevyengine/bevy/blob/main/crates/bevy_ecs/src/system/commands/mod.rs#L573)?
Probably easier to read. And this way we can make `SystemId` fields be private once again. (Not sure why they aren't pub to begin with)

Reply:
I don't understand what you mean.